src/libs/antares/study/variable-print-info.cpp

```diff
@@ -239,56 +222,58 @@ void AllVariablesPrintInfo::computeMaxColumnsCountInReports()
         Among all reports a study can create, which is the one that contains the largest
         number of columns and especially what is this number ?
         If there are some unselected variables, the previous number is reduced.
-        This number is a rough over-estimation, not the exact maximum number a report can
-        contain.
+        Note that synthesis reports always contain more columns than year by year reports.
+        So the computed max number of columns is actually the max number of columns in a synthesis report.
     */
 
-    uint CFileLevel = 1;
-    uint CDataLevel = 1;
-
     // Looping over all kinds of data levels (area report, link reports, districts reports, thermal
-    // reports,...) and file levels (values reports, years ids reports, details reports, ...) the
-    // code can produce. For one particular kind of report, looping over (selected) output variables
+    // reports,...) and for a given data level, looping over file levels 
+    // (values reports, years ids reports, details reports, ...) the
+    // code can produce. 
+    // For one particular kind of report, looping over (selected) output variables
     // it contains, and incrementing a counter with as many columns as the current variable can take
     // up at most in a report.
-    while (CDataLevel <= Category::maxDataLevel && CFileLevel <= Category::maxFileLevel)
-    {
-        uint currentColumnsCount = 0;
 
-        for (auto it = allVarsPrintInfo.begin(); it != allVarsPrintInfo.end(); it++)
+    for (uint CDataLevel = 1; CDataLevel <= Category::maxDataLevel; CDataLevel *= 2)
+    {
+        for (uint CFileLevel = 1; CFileLevel <= Category::maxFileLevel; CFileLevel *= 2)
         {
-            if ((*it)->isPrinted() && (*it)->getFileLevel() & CFileLevel
-                && (*it)->getDataLevel() & CDataLevel)
-                currentColumnsCount += (*it)->getMaxColumnsCount();
+            uint currentColumnsCount = 0;
+            for (auto& [name, variable] : allVarsPrintInfo)
+            {
+                if (variable.isPrinted() &&
+                    variable.getFileLevel() & CFileLevel &&
+                    variable.getDataLevel() & CDataLevel)
```

Comment:
```suggestion
                    variable.isFileLevel(CFileLevel) &&
                    variable.isDataLevel(CDataLevel))
```
Apply the `&` operator inside the method instead. Looks a bit more clear

Reply:
Done